src/components/sections/local-signer.tsx

```diff
@@ -0,0 +1,286 @@
+    const entryPoint = ENTRYPOINT_ADDRESS_V07
+    const publicClient = createPublicClient({
+      transport: http(),
+      chain: sepolia
+    })
+
+    const bundlerClient = createBundlerClient({
+      transport: http(bundlerUrl),
+      entryPoint: entryPoint,
+      chain: sepolia
+    }).extend(pimlicoBundlerActions(entryPoint))
+
+    const paymasterUrl = `https://api.pimlico.io/v2/sepolia/rpc?apikey=${apiKey}`
+
+    const { accountAddress, permissionValidatorAddress } = permissionContext
+
+    const testDappPrivateKey = signerPrivateKey! as `0x${string}`
+    const dappAccount = signer!
+    console.log('dappAccount(permitted Signer) Address: ', dappAccount.address)
+
+    const isModuleInstalled = await publicClient.readContract({
+      address: accountAddress,
+      abi: isModuleInstalledAbi,
+      functionName: 'isModuleInstalled',
+      args: [
+        BigInt(1), // ModuleType
+        permissionValidatorAddress, // Module Address
+        '0x' // Additional Context
+      ]
+    })
+    console.log(`isModuleInstalled : ${isModuleInstalled}`)
+
+    const nonce = await getNonceWithContext(publicClient, {
+      sender: accountAddress,
+      entryPoint,
+      permissionValidatorAddress
+    })
+    const actions = [{ target: to, value: value, callData: data }]
+    const callData = await getCallDataWithContext(permissionContext, actions)
+    console.log('PermissionValidator Nonce: ', nonce)
+
+    const gasPrice = await bundlerClient.getUserOperationGasPrice()
+    const userOp: UserOperation<'v0.7'> = {
+      sender: accountAddress,
+      nonce: nonce,
+      callData: callData,
+      callGasLimit: BigInt(2000000),
+      verificationGasLimit: BigInt(2000000),
+      preVerificationGas: BigInt(2000000),
+      maxFeePerGas: gasPrice.fast.maxFeePerGas,
+      maxPriorityFeePerGas: gasPrice.fast.maxPriorityFeePerGas,
+      signature: '0x'
+    }
+    console.log('Partial userOperation :', userOp)
+
+    const userOpHash = getUserOperationHash({
+      userOperation: {
+        ...userOp,
+        signature: '0x'
+      },
+      entryPoint: entryPoint,
+      chainId: sepolia.id
+    })
+    console.log('userOpHash : ', userOpHash)
+
+    const dappSignatureOnUserOp = await sign({
+      privateKey: testDappPrivateKey,
+      hash: userOpHash
+    })
+    const rawSignature = signatureToHex(dappSignatureOnUserOp)
+    console.log(rawSignature)
+
+    const finalSigForValidator = await getSignatureWithContext(rawSignature, permissionContext)
+
+    userOp.signature = finalSigForValidator
+    console.log(userOp)
+
+    const packedUserOp = getPackedUserOperation(userOp)
+    console.log(packedUserOp)
+
+    const _userOpHash = await bundlerClient.sendUserOperation({
+      userOperation: userOp
+    })
+
+    let txReceipt: GetUserOperationReceiptReturnType | null = null
+    while (txReceipt == null)
+      txReceipt = await bundlerClient.getUserOperationReceipt({
+        hash: _userOpHash
+      })
```

Comment:
How many requests are we sending here?

Reply:
only one to purchase donut, line58,

Reply:
it is trying to get the receipt for the sent userOps until it get the response back, I should add a delay of few sec btw the requests, or 
best approach would be to use `waitForUserOperationReceipt`  method in permissionless which will do this internally.

Reply:
Method in permissionless has a polling interval. Your code will send huge amount of requests until it gets a result. Its not the same. Lets stick with permissionless function.

Reply:
Yes yes, I meant do the same.

Reply:
Alright. Lets amend and merge after